File: typescript/src/01-strategy/after/razorpay-payment-strategy.ts
import type { IPaymentStrategy } from './payment-strategy.js';
import type { PaymentResult } from '../../shared/models.js';

export class RazorpayPaymentStrategy implements IPaymentStrategy {
  pay(amount: number): PaymentResult {
    console.log(`  [After] Charging ₹${amount} via Razorpay strategy`);
    return { success: true, message: 'Razorpay payment processed' };
  }
}

File: csharp/DesiArchitect.DesignPatterns/01_Strategy/After/RazorpayPaymentStrategy.cs
using DesiArchitect.DesignPatterns.Shared;

namespace DesiArchitect.DesignPatterns.Strategy.After;

public class RazorpayPaymentStrategy : IPaymentStrategy
{
    public PaymentResult Pay(decimal amount)
    {
        Console.WriteLine($"  [After] Razorpay payment of ₹{amount} processed via Razorpay API");
        return new PaymentResult(true, "Razorpay payment successful");
    }
}
